akka-docs/rst/java/futures.rst

```diff
@@ -260,3 +260,87 @@ After
 
 .. includecode:: code/docs/future/FutureDocTest.java
    :include: after
+
+Java 8, CompletionStage and CompletableFuture
+---------------------------------------------
+
+Starting with Akka 2.4.2 we have begun to introduce Java 8 ``java.util.concurrent.CompletionStage`` in Java APIs.
+It's a ``Future`` counterpart in Java; conversion from ``Future`` is done using ``scala-java8-compat`` library.
+
+Unlike ``Future``, ``CompletionStage`` has *async* and *non-async* methods.
```

Comment:
Well I actually meant "Unlike `scala.concurrent.Future` with only async methods, ..."
Because, AFAIK, `scala.concurrent.Future` always requires an (implicit) Executor.

So, Java `CompletionStage`: sync, async, async with an executor.
Scala Future: async with an executor.